idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -564,8 +567,13 @@ def run_host_filtering(fastq_files, initial_file_type_for_log, lazy_run, stats):
                       before_filetype="fasta_paired",
                       after_filename=os.path.join(RESULT_DIR, EXTRACT_UNMAPPED_FROM_SAM_OUT1),
                       after_filetype="fasta_paired")
+def init_stage1():
+    global STAR_BOWTIE_VERSION_FILE_S3
+    if not STAR_BOWTIE_VERSION_FILE_S3:
+        STAR_BOWTIE_VERSION_FILE_S3 = get_host_index_version_file()
```

Comment:
My concern here is that STAR_BOWTIE_VERSION_FILE_S3 is not initialized until this runs.   The better approach is to change the initializer for the global, like so:

STAR_BOWTIE_VERSION_FILE_S3 = os.getenv(STAR_BOWTIE_VERSION_FILE_S3, get_host_index_version_file())

That way any code that refers to this global will access the correctly initialized value.

Reply:
can't do that unless  I put some code before the constants setting. that's why i do init instead.